### PR TITLE
[FLINK-1923] Replaces asynchronous logging with synchronous logging in actors

### PR DESCRIPTION
**Pull request by tillrohrmann**
Replaces asynchronous logging with synchronous logging in actors. Additionally, all Scala implementations are now using the grizzled-slf4j SLF4J-wrapper which allows proper usage of SLF4J within Scala code.

One problem the grizzled-slf4j wrapper fixes is the ambiguity between varargs and a string with two placeholders. Additionally it resolves the ambiguity between (String, String, Object) where the first String is used as a Marker and where the first string is the logging message.

Grizzled-slf4j adds automatically logging guards which make the explicit checking for the log level redundant.

Grizzled-slf4j's license is BSD.

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java [hunk 1]
```diff
@@ -357,16 +356,6 @@ protected void unregisterTask() {
 		taskManager.tell(new UnregisterTask(executionId), ActorRef.noSender());
 	}
 
-	protected void notifyExecutionStateChange(ExecutionState executionState,
```
Comment (StephanEwen) [inline flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java]: Was this method unused?

**Comment by StephanEwen** [comment]
I am curious, why did you rewrite the TaskManager? I thought that one was logging synchronously already.

File: flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java (hunk 1 above)
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java]: Yes.

**Comment by tillrohrmann** [comment]
I changed the TaskManager logging to make it consistent with the way the other actors do the logging. Furthermore, by using a trait to mix the logger in, we can easily switch between asynchronous and synchronous logging. We could even make this configurable if the synchronous logging should become a performance issue.

**Comment by StephanEwen** [comment]
Okay, fair enough.

I'll merge this change in the next batch. Working on a change that would otherwise conflict...

**Pull request closed without merging**